src/lib.rs

```diff
@@ -53,11 +54,29 @@ fn parse_fields(item: &syn::DeriveInput, mutable: bool) -> (syn::Type, proc_macr
         _ => panic!("#[derive({})] can only be used on structs", trait_name),
     };
 
-    let field_ty = if fields.len() == 1 {
-        fields[0].ty.clone()
-    } else {
-        panic!("#[derive({})] can only be used on structs with one field", trait_name)
+    let field_ty = match fields.len() {
+        1 => Some(fields[0].ty.clone()),
+        2 => {
+            if let Type::Path(TypePath { path: Path { segments, .. }, .. }) = &fields[1].ty {
+                if segments
+                    .last()
+                    .expect("Expected path to have at least one segment")
+                    .value()
+                    .ident == "PhantomData"
+                {
+                    Some(fields[0].ty.clone())
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        },
+        _ => None,
     };
+    let field_ty = field_ty
+        .unwrap_or_else(|| panic!("#[derive({})] can only be used on structs with one field", trait_name));
```

Comment:
Maybe this should be changed to say that it allows `PhantomData` as a second field ?